src/busterbunny.js

```diff
@@ -305,6 +319,24 @@ module.exports = (function() {
             self.emit(self.EVENTS.READY);
         });
 
+        self.on(self.EVENTS.DISCONNECTING, function() {
+            try{
+                if(_connection && _connected){
+                    _connection.close(function(){});
+                    _connected = false;
+                }
+            }
+            catch(err)
+            {
+                _connected = false;
+            }
+            finally
+            {
+                self.emitStats();
```

Comment:
shouldn't this & the subsequent disconnected event only be emitted in the callback of close?